src/compas_timber/elements/fasteners/fastener.py

```diff
@@ -0,0 +1,164 @@
+from compas_model.elements import Element
+from compas_model.elements import reset_computed
+
+
+class Fastener(Element):
+    """
+    A class to represent timber fasteners (screws, dowels, brackets).
+
+    Parameters
+    ----------
+    elements : list(:class:`~compas_timber.parts.Element`)
+        The elements that are connected with this fastener.
+
+    Attributes
+    ----------
+    elements : list(:class:`~compas_timber.parts.Element`)
+        The elements that are connected with this fastener.
+
+    """
+
+    def __init__(self, elements, **kwargs):
+        super(Fastener, self).__init__(elements, **kwargs)
+        self.elements = elements
+        self.features = []
+        self.attributes = {}
+        self.attributes.update(kwargs)
+        self.debug_info = []
+
+    def __repr__(self):
+        # type: () -> str
+        element_str = ["{} {}".format(element.__class__.__name__, element.key) for element in self.elements]
+        return "Fastener({})".format(", ".join(element_str))
+
+    # ==========================================================================
+    # Computed attributes
+    # ==========================================================================
+
+    @property
+    def is_fastener(self):
```

Comment:
@chenkasirer can we have each element type register itself with the generic Element class so we don't need an explicit `is_fastener()` property in the Element class?